### PR TITLE
SDIT-1900 Prevent new bookings being created after an escape

### PR DESCRIPTION
**Pull request by andymarke**
Also mapped the escape flag for movements and experiment for LocalDateTime DSL for better readability

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/BookingIntoPrisonService.kt [hunk 1]
```diff
@@ -85,7 +85,9 @@ class BookingIntoPrisonService(
   }
 
   fun newBookingWithoutUpdateLock(offender: Offender, previousBooking: OffenderBooking?, requestForNewBooking: RequestForNewBooking): InmateDetail {
-    previousInactiveBooking(previousBooking).getOrThrow()
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/BookingIntoPrisonService.kt:88]: slight semantic change for readability

File: src/main/resources/db/migration/data/R__1_13__MOVEMENT_REASONS.sql [hunk 2]
```diff
@@ -1,7 +1,7 @@
 INSERT INTO MOVEMENT_REASONS (MOVEMENT_TYPE, MOVEMENT_REASON_CODE, DESCRIPTION, LIST_SEQ) VALUES ('REL',  'DD',      'Movement 1',   1);
-INSERT INTO MOVEMENT_REASONS (MOVEMENT_TYPE, MOVEMENT_REASON_CODE, DESCRIPTION, LIST_SEQ) VALUES ('REL',  'UAL',     'Movement 10', 2);
-INSERT INTO MOVEMENT_REASONS (MOVEMENT_TYPE, MOVEMENT_REASON_CODE, DESCRIPTION, LIST_SEQ) VALUES ('REL',  'UAL_ECL', 'Movement 11', 3);
-INSERT INTO MOVEMENT_REASONS (MOVEMENT_TYPE, MOVEMENT_REASON_CODE, DESCRIPTION, LIST_SEQ) VALUES ('REL',  'ESCP',    'Movement 12', 4);
+INSERT INTO MOVEMENT_REASONS (MOVEMENT_TYPE, MOVEMENT_REASON_CODE, DESCRIPTION, LIST_SEQ, ESC_RECAP_FLAG) VALUES ('REL',  'UAL',     'Abscond', 2, 'Y');
```
Comment (andymarke) [inline src/main/resources/db/migration/data/R__1_13__MOVEMENT_REASONS.sql:2]: set like production data (though UAL_ECL is no longer used

File: src/test/java/uk/gov/justice/hmpps/prison/dsl/DateHelper.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.hmpps.prison.dsl
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Suppress("ktlint:standard:function-naming")
+// add more months when required
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/dsl/DateHelper.kt:8]: silliness to avoid LocaDateTime.parse() to annoy @mikehalmamoj

Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/hmpps/prison/dsl/DateHelper.kt:8]: It's beautiful 😡

**Review by lbennett-moj — DISMISSED** [review]
lgtm

File: src/main/java/uk/gov/justice/hmpps/prison/exception/CustomErrorCodes.kt [hunk 4]
```diff
@@ -3,4 +3,5 @@ package uk.gov.justice.hmpps.prison.exception
 object CustomErrorCodes {
   const val NO_CELL_CAPACITY = 30001
   const val PRISONER_ALREADY_EXIST = 30002
+  const val PRISONER_ESCAPED_NEW_BOOKING_NOT_ALLOWED = 30002
```
[the remark this answers is not in the document]
Comment (andymarke) [reply src/main/java/uk/gov/justice/hmpps/prison/exception/CustomErrorCodes.kt:6]: oops - I will correct

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**